internal/component/azure/eventhubs/eventhubs.go

```diff
@@ -249,7 +336,24 @@ func (aeh *AzureEventHubs) Subscribe(subscribeCtx context.Context, topic string,
 	return nil
 }
 
-func (aeh *AzureEventHubs) processEvents(subscribeCtx context.Context, topic string, partitionClient *azeventhubs.ProcessorPartitionClient, eventHandler func(e *azeventhubs.ReceivedEventData) error) error {
+// Processes received eventhubs messages asynchronously
+func (aeh *AzureEventHubs) handleAsync(ctx context.Context, topic string, messages []*azeventhubs.ReceivedEventData, handler HandlerFn) error {
+	resp, err := handler(ctx, messages)
+	if err != nil {
+		// If we have a response with 0 items (or a nil response), it means the handler was a non-bulk one
+		if len(resp) == 0 {
+			aeh.logger.Errorf("Failed to process Eventhubs message %s for topic %s: Error: %v", messages[0].MessageID, topic, err)
+		}
+		for _, item := range resp {
+			if item.Error != nil {
+				aeh.logger.Errorf("Failed to process Eventhubs bulk message entry. EntryId: %s. Error: %v ", item.EntryId, item.Error)
+			}
+		}
+	}
+	return err
+}
+
+func (aeh *AzureEventHubs) processEvents(subscribeCtx context.Context, topic string, partitionClient *azeventhubs.ProcessorPartitionClient, maxBulkSubCount int, maxBulkSubAwaitDurationMs int, handler HandlerFn) error {
```

Comment:
I have added support for `checkpointFrequencyPerPartition` in the PR. It will basically checkpoint the messages with this frequency for every partition. Would that be good? cc @ItalyPaleAle

Reply:
I am not familiar with checkpoints enough, but code LGTM

Reply:
@shivamkm07 Given this is a pubsub abstraction, does the "partition" suffix make sense here given that other pubsub backends may not have partitions? Especially when you don't even allow per-partition settings (which is not something the native SDK allows anyway.)  tl;dr Leaky abstraction?